proof/invariant-abstract/DetSchedSchedule_AI.thy

```diff
@@ -822,8 +823,14 @@ locale DetSchedSchedule_AI =
     "\<And>t. \<lbrace>valid_list\<rbrace> arch_switch_to_thread t \<lbrace>\<lambda>_. valid_list\<rbrace>"
   assumes arch_switch_to_idle_thread_valid_list'[wp]:
     "\<lbrace>valid_list\<rbrace> arch_switch_to_idle_thread \<lbrace>\<lambda>_. valid_list\<rbrace>"
-  assumes prepare_thread_delete_idel_thread[wp] :
-    "\<And>t. prepare_thread_delete t \<lbrace>\<lambda>(s:: det_ext state). P (idle_thread s)\<rbrace>"
+  assumes prepare_thread_delete_idle_thread[wp]:
+    "\<And>P t. prepare_thread_delete t \<lbrace>\<lambda>s::det_state. P (idle_thread s)\<rbrace>"
```

Comment:
Nice typo fix. I'm also wondering why you've introduced `state_ext` here and used it for the stuff you've added, but the other ones use `det_state`

Reply:
We probably should have always had a `state_ext` and been using it where we could, but I think that we were lazy at the time due to most of the DetSched lemmas originally needing to be `det_state` anyway. This has changed since then and I'm using the ones I've added in other places where them being `det_state` causes problems.

Reply:
I think there's a comment on one of the DetSched locales saying something like this and that we could be using `state_ext` more.